ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/helpcenter/articlepage/ArticleFragmentTest.java

```diff
@@ -0,0 +1,514 @@
+    private View view;
+
+    @Mock
+    private LayoutInflater layoutInflater;
+
+    @Mock
+    private ViewGroup viewGroup;
+
+    @Mock
+    private DefaultStateViewHelper defaultStateViewHelper;
+
+    @Mock
+    private TextView textView;
+
+    @Mock
+    private CircleImageView circleImageView;
+
+    @Mock
+    private WebView webView;
+
+    @Mock
+    private WebSettings webSettings;
+
+    @Mock
+    private Intent intent;
+
+    @Mock
+    private Uri uri;
+
+    @Test
+    public void givenArticleWhenNewInstanceThenReturnFragment()
+            throws Exception {
+        //Arrange
+        mockStatic(ArticleFragment.class);
+        whenNew(ArticleFragment.class).withAnyArguments()
+                .thenReturn(articleFragment);
+        whenNew(Bundle.class).withAnyArguments().thenReturn(bundle);
+
+        //Act
+        ArticleFragment returnedValue = ArticleFragment
+                .newInstance(article);
+
+        //Assert
+        assertEquals(articleFragment, returnedValue);
+    }
+
+    @Test
+    public void givenBundleWhenOnCreateThenGetPresenter() {
+        //Arrange
+        mockStatic(ArticleFactory.class);
+        when(ArticleFactory.getPresenter(articleFragment))
+                .thenReturn(presenter);
+
+        //Act
+        articleFragment = new ArticleFragment();
+        articleFragment.onCreate(bundle);
+
+        //Assert
+        verifyStatic();
+        ArticleFactory.getPresenter(articleFragment);
+    }
+
+    @Test
+    public void givenLayoutViewGroupBundleWhenOnCreateViewThenReturnView()
+            throws Exception {
+        //Arrange
+        when(layoutInflater
+                .inflate(R.layout.ko__fragment_article_content, null))
+                .thenReturn(view);
+        whenNew(DefaultStateViewHelper.class).withAnyArguments()
+                .thenReturn(defaultStateViewHelper);
+        articleFragment = new ArticleFragment();
+
+        //Act
+        View returnedValue = articleFragment
+                .onCreateView(layoutInflater, viewGroup, bundle);
+
+        //Assert
+        assertEquals(view, returnedValue);
+    }
+
+    @Test
+    public void givenViewBundleWhenOnViewCreatedThenPresenterInitPage() {
+        //Arrange
+        ArticleFragment articleFragmentSpy =
+                spy(new ArticleFragment());
+        articleFragmentSpy.setArguments(bundle);
+        when(bundle.getSerializable(ARG_ARTICLE)).thenReturn(article);
+        mockStatic(ArticleFactory.class);
+        when(ArticleFactory.getPresenter(articleFragmentSpy))
+                .thenReturn(presenter);
+        articleFragmentSpy.onCreate(bundle);
+
+        //Act
+        articleFragmentSpy.onViewCreated(view, bundle);
+
+        //Assert
+        verify(presenter).initPage(article);
+    }
+
+    @Test
+    public void givenStringWhenSetAuthorNameThenSetText()
+            throws Exception {
+        //Arrange
+        ArticleFragment articleFragmentSpy =
+                spy(new ArticleFragment());
```

Comment:
⚠️ It might be a good idea break lines only when there is more than one statement in the expression or when the line has too many characters (>120). Consider the proposed solution.

## The possible solution for this case

```Java
ArticleFragment articleFragmentSpy = spy(new ArticleFragment());
``` 

<a id="1b96c910-4e5a-11e8-860c-69c5dc5809f4" href="none" codereview-uuid-reserved-tag />